.github/workflows/prepare-test-config.yaml

```diff
@@ -0,0 +1,130 @@
+
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.overlay-branch }}
+          repository: ${{ inputs.overlay-repo }}
+
+      - name: Download last published-exports artifact
+        uses: dawidd6/action-download-artifact@v6
+        with:
+          workflow: pr-actions.yaml
+          name: published-exports
+          workflow_conclusion: success
+          workflow_search: true
+          search_artifacts: true
+          check_artifacts: true
+          if_no_artifact_found: warn
+          allow_forks: true
+
+      - name: Parse config and update OCI references
+        run: |
+          WORKSPACE_PATH="${{ inputs.workspace-path }}"
+          PUBLISHED_EXPORTS=""
+
+          # Fallback: read from downloaded artifact if input was empty
+          if [ -f "published-exports.txt" ]; then
+            PUBLISHED_EXPORTS="$(cat published-exports.txt)"
+            echo "Artifact content: $PUBLISHED_EXPORTS"
+          fi
+
+          if [ -z "$PUBLISHED_EXPORTS" ]; then
+            echo "No published exports provided or found via artifact. Nothing to do."
+            exit 0
+          fi
+
+          OUT_DIR="$WORKSPACE_PATH/tests"
+          OUT_FILE="$OUT_DIR/dynamic-plugins.test.yaml"
+          mkdir -p "$OUT_DIR"
+
+          # Always include the root-level default config
+          ROOT_CONFIG="$GITHUB_WORKSPACE/tests/app-config.yaml"
+          if [ -f "$ROOT_CONFIG" ]; then
+            echo "Copying root app-config.yaml into artifact"
+            cp "$ROOT_CONFIG" "$OUT_DIR/app-config.yaml"
+          else
+            echo "Root-level tests/app-config.yaml not found at $ROOT_CONFIG"
+          fi
+
+          # copy optional workspace-level test configs into artefact
+          [ -f "$WORKSPACE_PATH/tests/app-config.test.yaml" ] && cp "$WORKSPACE_PATH/tests/app-config.test.yaml" "$OUT_DIR/"
+          [ -f "$WORKSPACE_PATH/tests/test.env" ] && cp "$WORKSPACE_PATH/tests/test.env" "$OUT_DIR/"
+
+          echo "Contents of $OUT_DIR after copy:" && ls -la "$OUT_DIR"
+
+          # Install yq for YAML processing
+          sudo curl -sL https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64 -o /usr/local/bin/yq
+          sudo chmod +x /usr/local/bin/yq
+
+          # Start the resulting YAML file
+          echo "plugins:" > "$OUT_FILE"          
+
+          # For each published export, extract plugin name and read its metadata
+          # Expected export format: ghcr.io/<repo_path>/<plugin_name>:<tag>
+          for export in $PUBLISHED_EXPORTS; do
+            echo "Processing export: $export"
+            if [[ "$export" =~ ^ghcr\.io/(.+):([^[:space:]]+)$ ]]; then
+              IMAGE_PATH_AND_PLUGIN="${BASH_REMATCH[1]}"   # <repo_path>/<plugin_name>
+              NEW_TAG="${BASH_REMATCH[2]}"                 # <tag>
+              PLUGIN_NAME="${IMAGE_PATH_AND_PLUGIN##*/}"
+
+              METADATA_FILE="$WORKSPACE_PATH/metadata/${PLUGIN_NAME}.yaml"
```

Comment:
I'm not sure it would be guaranteed that the name of the yaml file in the `metadata` folder would always be exactly the same as the one extracted from the image name.
But re-creating the container image name from the package name contained in the metadata yaml and using that to match metadata yaml files with OCI references seems more robust.

Reply:
(because we know the logic of the oci reference name from the package name, so that we can reproduce it here.)